### ISSUE TITLE
Roles and permissions.

### ISSUE THREAD

**Issue by italoantonio0909** [HackSoftware/Django-Styleguide-Example#103]
Hello, some time ago I followed this style guide, I find it interesting, permissions and roles should live in a module?, or extract logic in the same user module?

**Comment by RadoRado** [comment]
@italoantonio0909 Hello :wave: 

Generally speaking, we do the following:

1. We don't use Django's roles & permissions, unless we want to model something very specific related to Django admin (for example, there's no time for a proper backoffice implemetation / integration, end users are going to be using the Django admin (which actually does the job better than many other backoffice products) and we need to restrict a lot of things.
2. Roles and permissions live on their own, with their very specific implementation, usually in an app called `permissions`

Here's a very simple model structure, that gives you:

1. Groups with members there
2. Permissions related to those groups

```python
from django.db import models

from django_enum_choices.fields import EnumChoiceField

from project.common.models import BaseModel
from project.users.models import BaseUser

from project.permissions.constants import Action


class AdminGroup(BaseModel):
    name = models.CharField(max_length=255, unique=True)

    members = models.ManyToManyField(
        BaseUser,
        related_name='admin_groups',
        limit_choices_to={'is_admin': True}
    )

    def __str__(self):
        return self.name


class AdminGroupPermission(BaseModel):
    group = models.ForeignKey(
        AdminGroup,
        related_name='permissions',
        on_delete=models.CASCADE
    )

    action = EnumChoiceField(Action)

    value = models.BooleanField(default=False)

    def __str__(self):
        return f'{self.action.value} for {self.group}'

    class Meta:
        unique_together = (('group', 'action'))
```

And now you can create your own groups & give specific permissions to those groups & then members belonging to those groups can do those specific actions.

So long story short - we implement our own groups, roles & permissions, based on the project context.

**Comment by RadoRado** [comment]
@italoantonio0909 I'm closing this issue for now. Feel free to reopen it, in case you have any questions :raised_hands:
